web/src/js/__tests__/ducks/modes/localSpec.tsx

```diff
@@ -0,0 +1,147 @@
+    });
+
+    it("should dispatch MODE_LOCAL_TOGGLE and updateMode", async () => {
+        enableFetchMocks();
+
+        const store = TStore();
+
+        expect(store.getState().modes.local.active).toBe(false);
+        await store.dispatch(toggleLocal());
+        expect(store.getState().modes.local.active).toBe(true);
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it("should dispatch MODE_LOCAL_SET_APPLICATIONS and updateMode", async () => {
+        enableFetchMocks();
+        const store = TStore();
+
+        await store.dispatch(setApplications("curl"));
+
+        const state = store.getState().modes.local;
+        expect(state.applications).toEqual("curl");
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it("should handle RECEIVE_OPTIONS action", () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["local:curl"],
+                },
+            },
+        };
+        const newState = localReducer(undefined, action);
+        expect(newState.active).toBe(true);
+        expect(newState.applications).toBe("curl");
+        expect(newState.error).toBeUndefined();
+    });
+});
+
+describe("getMode", () => {
+    it("should return local mode with applications when active and applications are present", () => {
+        const modes = {
+            local: {
+                active: true,
+                applications: "curl",
+            },
+        };
+        const result = getMode(modes);
+        expect(result).toEqual(["local:curl"]);
+    });
+
+    it("should return local mode without applications when active and no applications are present", () => {
+        const modes = {
+            local: {
+                active: true,
+                applications: "",
+            },
+        };
+        const result = getMode(modes);
+        expect(result).toEqual(["local"]);
+    });
+
+    it("should return an empty array when local mode is not active", () => {
+        const modes = {
+            local: {
+                active: false,
+                applications: "curl",
+            },
+        };
+        const result = getMode(modes);
+        expect(result).toEqual([]);
+    });
+
+    it("should return an empty array when local mode is defined but active is false", () => {
+        const modes = {
+            local: {
+                active: false,
+                applications: "",
+            },
+        };
+        const result = getMode(modes);
+        expect(result).toEqual([]);
+    });
+
+    it("should return local mode with multiple applications when active and multiple applications are present", () => {
+        const modes = {
+            local: {
+                active: true,
+                applications: "curl, wget",
+            },
+        };
+        const result = getMode(modes);
+        expect(result).toEqual(["local:curl, wget"]);
+    });
```

Comment:
I do not mind these two tests here, but if you are comfortable with this you can skip such tests if you want. We already have 100% line (and branch) coverage with the tests above, which is more than good enough in my book. 😉